From d48649fcf94c733164504bfd61c47094ac440310 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sat, 15 Feb 2020 23:12:30 +0200
Subject: [PATCH] [SW-8229] goya2: fix disable debug interface

All interfaces have to override configuration to enable shared.
This garanties that the range registers will affect this interface.

Change-Id: I226aab9b2f562923291b812df06f334e64f98f9e
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 8565b5b..95a9d96 100644
--- a/common/security.c
+++ b/common/security.c
@@ -13,6 +13,9 @@
 
 static void security_disable_i2c_s_mstr_if(void)
 {
+	/* set AXUSER to shared */
+	regmap_writel(I2C_S_MSTR_IF_AXUSER_CORE_SHARED_MASK,
+	      mmI2C_S_MSTR_IF_AXUSER_CORE);
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
@@ -28,6 +31,8 @@
 
 static void security_disable_smi_mstr_if(void)
 {
+	regmap_writel(SMI_MSTR_IF_AXUSER_CORE_SHARED_MASK,
+	      mmSMI_MSTR_IF_AXUSER_CORE);
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
@@ -43,6 +48,9 @@
 
 static void security_disable_jtag_mstr_if(void)
 {
+	regmap_writel(JT_MSTR_IF_AXUSER_CORE_SHARED_MASK,
+		      mmJT_MSTR_IF_AXUSER_CORE);
+
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
